corehq/apps/users/views/__init__.py

```diff
@@ -704,10 +705,15 @@ def create_domain_permission_mirror(request, domain):
         mirror_domain_name = form.cleaned_data.get("mirror_domain")
         mirror_domain = Domain.get_by_name(mirror_domain_name)
         if mirror_domain is not None:
-            mirror = DomainPermissionsMirror(source=domain, mirror=mirror_domain_name)
-            mirror.save()
-            message = _('You have successfully added the project space "{mirror_domain_name}".')
-            messages.success(request, message.format(mirror_domain_name=mirror_domain_name))
+            try:
+                mirror = DomainPermissionsMirror(source=domain, mirror=mirror_domain_name)
+                mirror.save()
+            except IntegrityError:
```

Comment:
nit: this check might make more sense in `clean_mirror_domain` as part of form validation.

Reply:
actually same with the `if mirror_domain is not None` check above, though i know you didn't add that.  By moving them there, the errors would show up attached to the input field, which also seems nicer.

Reply:
I see what you mean. I'll look more at swapping those checks over tomorrow.

Reply:
I moved those error checks over to clean_mirror_domain. It seems to work with no issues when I test it locally, but if you want to take another peak I've pushed up the changes.